### DISCUSSION TITLE
YNAB Importer Button is Missing

### DISCUSSION THREAD

**Discussion by thomasgeorge** [firefly-iii/firefly-iii#4464] [Firefly III]
The "START" button on YNAB Importer is missing. It just doesn't display.
![image](https://user-images.githubusercontent.com/5393484/109756017-1960bf80-7bb5-11eb-8693-31b75c915b05.png)

**Comment by JC5** [comment]
It seems to be missing the CSS as well.

**Reply by thomasgeorge** [reply]
Correct

**Reply by thomasgeorge** [reply]
I was able to move forward by inspecting the element and navigating to the right page. But, now I'm stuck importing transactions after relating ynab to firefly

**Reply by JC5** [reply]
I suggest you tell us a bit how you installed the importer. I bet there is a difference in the URL from your browser, and the URL to the CSS files in the source.

**Reply by thomasgeorge** [reply]
I installed this within my firefly-iii folder, using the terminal/composer commands.
As such, YNAB importer is accessed at sub.domain.com/ynab-importer/public
firefly-iii is at sub.domain.com/public

**Reply by JC5** [reply]
Try setting the `ASSET_URL` to /ynab-importer/public`, it may work!

**Reply by thomasgeorge** [reply]
which file is that set at? pardon my newb, I can't find that setting in either .env file and not easily in any of the files I looked at

**Reply by JC5** [reply]
You can add it to the .env file yourself. Your milage may vary, installing tools inside tools is not somethign I'd recommend.

**Reply by thomasgeorge** [reply]
I sincerely appreciate all of your help.

Something's still off. I've tried reinstalling ynab-importer in the same parent folder as firefly-iii as well as adding various combinations of URLs as ASSET_URL, to no help.
They also have individual subdomains (ynab.domain and budget.domain) just to simplify this a bit

I'm surprised I even had firefly-iii up and running so easily (owing to my newbiness, not your documentation - it's great!) but I'm fairly stuck here. I understand if you can't help further. I truly appreciate it thus far.

**Reply by JC5** [reply]
Make no mistake, I love to help but at some point i'll be out of my depth re: your setup.

If they have individual subdomains it should work already, especially when the root is at / and the host is correct. FF3 and the YNAB importer use the same framework.

**Reply by thomasgeorge** [reply]
Thank you very much so far! 

They're definitely in different folders and on separate subdomains. They also have ssl certificates. Could you let me know all the scripts that need to be referenced from the main page? I was able to change get the css styling by hardcoding the full domain into the oage in developer tools in my browser.
Perhaps I could do the same within the files themselves? 

Alternatively, I'm happy to give you a door into my setup to see what's not installed correctly. 

Again, no pressure. I'll keep trying to get this to work for a couple more days probably. My next alternative would be to directly enter all  transactions into the database directly as that can be done in a somewhat progrmamatic fashion.

**Reply by thomasgeorge** [reply]
To be clear on the actual original problem, I can get as far as the "Configure your YNAB Import" screen, without changing anything. 

So if it's just CSS that's missing, that'd be no problem. 

In this case,however, it doesn't seem like the transactions get imported despite the importer being able to retrieve both my YNAB accounts as well as my FF3 account through their respective API's. 
This is the actual issue.

**Reply by thomasgeorge** [reply]
I'd also like to point out that the download/status?identifier goes from "waiting_to_start" to "job_done" so it's obviously doing something. Just doesn't get written to the database

**Reply by thomasgeorge** [reply]
The import/mapping does have an error 500. Happy to provide the link to the file to diagnose if you'd like. I'm digging around there myself.

**Reply by JC5** [reply]
At one point, the importer will need a specific JS file which it probably can't load either. If the CSS doesn't work the import will fail because the JS will also fail to load.

If you have a stack trace for the 500 error, hit me up: james@firefly-iii.org. It may contain private data.
